opentracing/tracecontext.go

```diff
@@ -49,18 +49,20 @@ type TraceContext interface {
 	// HTTP headers are case insensitive.
 	//
 	// `restrictedKey` MUST match the regular expression
-	// `(?i:[a-z0-9][-a-z0-9]*)` and is case-insensitive. See
-	// CanonicalizeTraceTagKey().
+	// `(?i:[a-z0-9][-a-z0-9]*)` and is case-insensitive. That is, it must
```

Comment:
@adriancole I think we're specifically concerned with comparison. The storage format can be implementation specific: lower case, or capitalized As-Many-Http-Libraries-Do, or whatever. The main contract is that if you set an attribute with key="ABC-xyz", you can retrieve it with key="abc-Xyz".